sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared/response_handlers.py

```diff
@@ -67,7 +67,7 @@ def normalize_headers(headers):
 
 
 def deserialize_metadata(response, obj, headers):  # pylint: disable=unused-argument
```

Comment:
Check on usage in datalake

Reply:
So I'm not exactly sure when this `deserialize_metadata` [method](https://github.com/Azure/azure-sdk-for-python/blob/e21e984c0752cd05ab51d66940afb1c0b9b6abbf/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared/response_handlers.py#L69) would be called as the only instance I see of metadata serialization is in `get_file_properties()` or `get_directory_properties()`. 


get_file_properties: `return self._get_path_properties(cls=deserialize_file_properties, **kwargs)`

Following this debugger trail will take you to `_deserialize.py`'s `deserialize_file_properties` method, which in turn will call its `deserialize_metadata` (and **NOT** the one defined in `response_handlers.py` found [here](https://github.com/Azure/azure-sdk-for-python/blob/e21e984c0752cd05ab51d66940afb1c0b9b6abbf/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py#L110)

The same is true for directories (will call into `_deserialize.py`'s methods and **NOT** `response_handlers.py`'s methods.

Furthermore, looking through datalake's imports from `from ._shared.response_handlers`, shows that none of them ever import `deserialize_metadata` explicitly (and no methods that _are_ imported from this file call `deserialize_metadata` as a subroutine. 

**tl;dr** - This is **unused**, so we can change it to keep _shared consistent, and if it does start being used it will error out and so all of them will be revisited to fix functionality + be the same across all packages

Reply:
In contrast, blob's `get_properties` call does call into `response_handlers.py`'s `deserialize_metadata` method, and this is the response object (hence why it needs .http_response):
![image](https://user-images.githubusercontent.com/101599632/210671206-b96843e9-9be6-40db-9d0a-0f644182aacd.png)

This is an example of datalake's response object:
![image](https://user-images.githubusercontent.com/101599632/210671301-644ebfaa-6657-4b7d-b4ee-8dde3f419454.png)

So blob's is a `PipelineResponse` which contains a `RequestsTransportResponse` while datalake is just the `RequestsTransportResponse`

Reply:
Take more robust one from `deserialize.py` and slap it into `_shared` (this change needs to be done in dlake & queue)